### PR TITLE
fix: add flag to hide geocoding validation by radius

### PR DESCRIPTION
**Pull request by KrissDrawing**
# Pull Request Template

## Issue Overview

This PR addresses #3948 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Adds feature flag that is default to `true` (so full geocoding functionality). On `false` it hides `"validate radius"` option, and on submit it removes it's fields (as there is possibility to create option on one jurisdiction (with radius validation) and then switch to other with pre-populated fields)

## How Can This Be Tested/Reviewed?

Switch flag to false for jurisdiction of choice. 
- in partners settings create/edit preference.
- Add option with jurisdiction of your choice (it should not have validate by radius radio box).
- Create options for jurisdiction that has flag on `true` then switch for jurisdiction of choice (with false) and submit -> it should switch all validationRadius to `will check manually`.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/prisma/migrations/09_add_geocoding_flag_radius_method/migration.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- AlterTable
+ALTER TABLE "jurisdictions" ADD COLUMN     "enable_geocoding_radius_method" BOOLEAN NOT NULL DEFAULT true;
```
Comment (mcgarrye) [inline api/prisma/migrations/09_add_geocoding_flag_radius_method/migration.sql:2]: Can we change the folder prefix to 10 as 09 is already in use?

**Review by mcgarrye — CHANGES_REQUESTED** [review]
Two things:
1. I feel like we should be defaulting to `false`. Based on the ticket, we want this turned off in all existing jurisdictions until further notice. So we should only be switching it to true on a per case basis going forward.

2. With the flag turned to true, I was able to see the field when editing existing option but NOT when creating new options. Not familiar with this code but maybe it is missing from the new option flow?

**Comment by KrissDrawing** [comment]
@mcgarrye thanks for the review!

1. Yes, it probably makes more sense to default it to `false`.
2. Not sure if it's what i think it is, but this flow is slightly counterintuitive. As when you create new `preference` then you have no jurisdiction selected (so no radius option present in `options`). Then when you select jurisdiction that has that flag turned on the option will appear. Only thing that i can improve here is to add this option initially (so when no jurisdiction selected) when every jurisdiction has flag turned on. Let me know if thats the issue, or is it something else 🤔

**Comment by mcgarrye** [comment]
> @mcgarrye thanks for the review!
> 
> 1. Yes, it probably makes more sense to default it to `false`.
> 2. Not sure if it's what i think it is, but this flow is slightly counterintuitive. As when you create new `preference` then you have no jurisdiction selected (so no radius option present in `options`). Then when you select jurisdiction that has that flag turned on the option will appear. Only thing that i can improve here is to add this option initially (so when no jurisdiction selected) when every jurisdiction has flag turned on. Let me know if thats the issue, or is it something else 🤔

In response to #2, that is definitely what I am seeing. If the user selects a jurisdiction before adding an option it does appear properly. I would just flag it to product but I think it is fine to leave it as is since all jurisdictions will have it turned off for now.

File: api/prisma/seed-helpers/jurisdiction-factory.ts [hunk 2]
```diff
@@ -17,5 +17,6 @@ export const jurisdictionFactory = (
   enableAccessibilityFeatures: true,
   enableUtilitiesIncluded: true,
   enableGeocodingPreferences: true,
+  enableGeocodingRadiusMethod: true,
```
Comment (mcgarrye) [inline api/prisma/seed-helpers/jurisdiction-factory.ts:20]: Can we flip this to false too please

Comment (KrissDrawing) [reply api/prisma/seed-helpers/jurisdiction-factory.ts:20]: @mcgarrye Do you have any specific reason for that?
It seem like locally it's not a problem to have it switched on.
It also fails tests (which i can change, but probably they will never appear later 😄 ).
So just let me know it you see any problems with that beeing `true` here.

Comment (mcgarrye) [reply api/prisma/seed-helpers/jurisdiction-factory.ts:20]: Generally we want the seed data to match the actual environment to the best of our ability. And would be good to have a test around the flag

Comment (KrissDrawing) [reply api/prisma/seed-helpers/jurisdiction-factory.ts]: It makes sense! I switched it to `false`, and modified test to use second option (so map layer validation). Is that ok ?

**Review by mcgarrye — APPROVED** [review]
As long as product is aware of the new setting flow when the flag is true that we discussed, this looks good to go. lgtm

**Review by YazeedLoonat — APPROVED** [review]
LGTM 🍝

**Pull request merged**